### PR TITLE
feature: search results first iteration

### PR DESCRIPTION
**Pull request by joshuagraber**
This page and its design is still a work in progress. This PR gets the basic search results functionality down. We'll fine-tune it further later.

## To test
1. run the app locally
2. Search
3. Observe search results page

### CODE REVIEW

**Review by josh-chamberlain — APPROVED** [review]
hmm, when I try to search, the `record type` part of the form just goes blank and I don't progress further. I'm willing to just assume this is a local issue and get this into `dev`.

**Comment by joshuagraber** [comment]
Hey @josh-chamberlain I'm not sure what you're referring to by the "record type" part. Are you talking about the checkboxes? If so, I'm not seeing this:


https://github.com/user-attachments/assets/9ed00407-9a02-4103-bba4-ad8745526887

**Comment by maxachis** [comment]
@joshuagraber Can confirm the same thing happens to me as well. Checking the developer console, I note I'm seeing 
```
Path from search:
 /search?state=Pennsylvania&county=Allegheny&locality=Pittsburgh&record_categories=Police+%26+public+interactions```

Which suggests to me that the version I'm looking at in the branch is the not-yet-updated version.

**Comment by maxachis** [comment]
Gave it a run! Here are my current results. Let me know if anything seems unclear from my data


https://github.com/user-attachments/assets/b5f09115-b39b-483f-80ba-6ac073f77602

**Comment by maxachis** [comment]
> hmm, when I try to search, the `record type` part of the form just goes blank and I don't progress further. I'm willing to just assume this is a local issue and get this into `dev`.

@josh-chamberlain I suggest double-checking that you've got the `feature/search-results` branch updated and checked out (which was my initial error). If you did all that and still got that error (which is separate from my own), then that's interesting.

@joshuagraber Do we currently have a system that runs Selenium (or comparable) tests on a spun-up, containerized version of the web app, similar in fashion to my roster of automated unit and integration tests?

**Comment by joshuagraber** [comment]
@josh-chamberlain @maxachis Ya'll just need to install updated dependencies. From `/client` run `npm i`

> Do we currently have a system that runs Selenium (or comparable) tests on a spun-up, containerized version of the web app, similar in fashion to my roster of automated unit and integration tests?

No, but I'll be adding integration test in [this issue](https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/377). We'll run those in CI on every pull request. 

I'm not containerizing anything locally, because containers really only make sense to me in this context if they're pushed to version control, so the env is the same for everyone, as well as in deployment.

**Comment by josh-chamberlain** [comment]
@joshuagraber I did check that I was on the correct branch with `npm i` having been run! 

screencast 2024-08-29 12-28-23

**Comment by joshuagraber** [comment]
@josh-chamberlain I still cannot reproduce any of this. Even if I could reproduce that `label` error, that wouldn't block the navigation.

https://github.com/user-attachments/assets/cb6c3454-4457-4b07-8d11-012dc6833298


What version of `node` are you on? And do you have any dependencies symlinked locally that may still be there?

**Comment by josh-chamberlain** [comment]
@joshuagraber node `v22.4.1`. The typeahead endpoint is working for me, but I'm getting 401s from the API; I believe my database connection is faulty. It is shockingly difficult for me to maintain a working local env, and I don't want it to hold us up.

**Comment by joshuagraber** [comment]
Gotcha @josh-chamberlain I'm guessing there's some env variable stuff, then.

**Comment by josh-chamberlain** [comment]
@joshuagraber Unfortunately I'm now getting that search bug on dev: https://data-sources-v2.pdap.dev/

**Comment by maxachis** [comment]
> @maxachis do you think it could be a credential issue?

tl;dr: Yes, and I fixed it!

@josh-chamberlain In fact I can confirm that it is at least partially a credential issue! I tried to run a search just now, and got a 401 (meaning Unauthorized Error). Meanwhile, looking at the request the website sent, I spied the following:

image

Note the last line which says "Based undefined". It looks like where the authorization was supposed to be provided was not. 

After looking in the code and espying the following:

```
const HEADERS_BASIC = {
	authorization: `Basic ${import.meta.env.VITE_ADMIN_API_KEY}`,
	'Content-Type': 'application/json',
};
const SEARCH_URL = `${import.meta.env.VITE_VUE_API_BASE_URL}/search/search-location-and-record-type`;
```

I realized that there was a new environment variable, `VITE_ADMIN_API_KEY`, which needed to be set! 

Now bear in mind that the `data-sources-v2.pdap.dev` website is connected to the stage database, so API keys which are used for the Test environment (where all users were wiped and a set of test users with test API keys were provided) won't be valid. So I instead used Marty's old API key which is still hovering around in the Stage environment (which is a mirror of the Production Database code).

That's a confusing chain of events (and one I should put effort into simplifying). But the point is I created a new `VITE_ADMIN_API_KEY` variable, plugged in Marty's old API key, et voila!

image

We be functional!

**Comment by maxachis** [comment]
@joshuagraber @josh-chamberlain We'll need to update the `README.md` file to mention the presence of this `VITE_ADMIN_API_KEY`, along with a description of what it does!

**Comment by joshuagraber** [comment]
Hey @josh-chamberlain @maxachis Yeah, my bad. I should have set this in DO when I merged the auth PR previously. I can get a PR up to add explanation to the `README`

> Now bear in mind that the data-sources-v2.pdap.dev website is connected to the stage database, so API keys which are used for the Test environment (where all users were wiped and a set of test users with test API keys were provided) won't be valid. So I instead used Marty's old API key which is still hovering around in the Stage environment (which is a mirror of the Production Database code).

> That's a confusing chain of events (and one I should put effort into simplifying). But the point is I created a new VITE_ADMIN_API_KEY variable, plugged in Marty's old API key, et voila!

Definitely agreed that this needs some simplifying. We don't want to have to either 1. write a script for local dev just to get an API key that works, or 2. have to manually add a new key to the dev deployment env every day.

**Comment by josh-chamberlain** [comment]
@joshuagraber thank you! That reminds me that you might want to check our thinking on this issue: https://github.com/Police-Data-Accessibility-Project/data-sources-app/issues/388

**Comment by joshuagraber** [comment]
PR for the README update is here: https://github.com/Police-Data-Accessibility-Project/data-sources-app-v2/pull/98
